OpenRA.Mods.Cnc/Activities/VoxelHarvesterDockSequence.cs

```diff
@@ -45,10 +45,15 @@ public override void OnStateDock(Actor self)
 
 		public override void OnStateUndock(Actor self)
 		{
-			dockingState = DockingState.Wait;
-
-			if (spriteOverlay != null && !spriteOverlay.Visible)
+			// If body.Docked wasn't set, we didn't actually dock and have to skip the undock overlay
+			if (!body.Docked)
+			{
+				dockingState = DockingState.Complete;
+				return;
```

Comment:
The `return` statement here is superfluous since there is no code behind it anyway. (The rest of the method are in `else` branches.)